examples/external-mysql/values.yaml

```
@@ -0,0 +1,6 @@
+spire-server:
+  dataStore:
+    sql:
+      plugin_data:
+        database_type: "mysql"
+        connection_string: "spire:CHANGEME@tcp(mysql:3306)/spire?parseTime=true"
```

```suggestion
        host: mysql
        port: 3306
        database: spire
        username: spire
        password: CHANGEME
        options:
          - parseTime: true
```

Too support the confimap combination with secrets we might want to more stronglytype the values file.

---

What happens when they add a new database type that isn't sqllite or a sql server? say, mongo or some other nosql database? Are we backing ourselves into a corner trying to interpret connection_string?

If we're adding config options I don't think they should be under plugin_data, which should match closely what is in spire's config. We probably should move it up a level to under sql: if we do that.

---

In that case we probably have to add another implementation that builds the connection string based on that database type

---

I guess in the case of internal databases, we can just use enable to flag if we totally ignore the config there. I'll take a stab at it.